flake.nix

```diff
@@ -0,0 +1,71 @@
+{
+  description = "Rust development environment";
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+        # Read the file relative to the flake's root
+        overrides = (builtins.fromTOML (builtins.readFile (self + "/rust-toolchain.toml")));
+        libPath = with pkgs; lib.makeLibraryPath [
+          # load external libraries that you need in your rust project here
+	    pkgs.openssl
+            pkgs.stdenv.cc.cc.lib
+        ];
+      in
+      {
+        devShells.default = pkgs.mkShell rec {
+          nativeBuildInputs = [ pkgs.pkg-config ];
+          buildInputs = with pkgs; [
+            clang
+            llvmPackages.bintools
+            rustup
+	    openssl
+            stdenv.cc.cc.lib
+          ];
+
+          RUSTC_VERSION = overrides.toolchain.channel;
+          
+          # https://github.com/rust-lang/rust-bindgen#environment-variables
```

Comment:
I think this and further magic you do to make bindgen work nice can be achieved by just adding `pkgs.rustPlatform.bindgenHook` to build inputs